src/Authenticators/LDAPAuthenticator.php

```diff
@@ -163,18 +166,39 @@ public static function authenticate($data, Form $form = null)
      * Try to authenticate using the fallback authenticator.
      *
      * @param array $data
-     * @param null|Form $form
+     * @param HTTPRequest $request
      * @return null|Member
+     * @internal param null|Form $form
+     */
+    protected function fallbackAuthenticate($data, HTTPRequest $request)
+    {
+        $authenticatorClass = Config::inst()->get(self::class, 'fallback_authenticator_class');
+        if ($authenticator = Injector::inst()->get($authenticatorClass)) {
+            $result = call_user_func(
+                [
+                    $authenticator,
+                    'authenticate'
+                ],
+                $data,
+                $request
+            );
+            return $result;
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getLoginHandler($link)
+    {
+        return LDAPLoginHandler::create($link, $this);
+    }
+
+    /**
+     * @inheritdoc
      */
-    protected static function fallback_authenticate($data, Form $form = null)
+    public function supportedServices()
     {
-        return call_user_func(
-            [
-                Config::inst()->get(self::class, 'fallback_authenticator_class'),
-                'authenticate'
-            ],
-            array_merge($data, ['Email' => $data['Login']]),
-            $form
-        );
+        return Authenticator::LOGIN | Authenticator::LOGOUT;
```

Comment:
I'm reasonably sure we need to support changing passwords too, but I suppose that we could log that and address it separately

Reply:
You're right!
I must check if the below is true and that bit too 👍 
```
SilverStripe\ActiveDirectory\Services\LDAPService:
  allow_password_change: true
```

Reply:
I feel like this method should just have the bitmask in it, we can put the conditional logic into a more specific method for changing passwords, otherwise the way this module represents itself is not consistent depending on the user's settings. What do you think?